codes/go/chapter_graph/graph_adjacency_list.go

```diff
@@ -49,62 +39,70 @@ func (g *graphAdjList) size() int {
 }
 
 /* 添加边 */
-func (g *graphAdjList) addEdge(vet1 vertex, vet2 vertex) {
+func (g *graphAdjList) addEdge(vet1 Vertex, vet2 Vertex) {
 	_, ok1 := g.adjList[vet1]
 	_, ok2 := g.adjList[vet2]
 	if !ok1 || !ok2 || vet1 == vet2 {
 		panic("error")
 	}
 	// 添加边 vet1 - vet2, 添加匿名 struct{},
-	g.adjList[vet1][vet2] = struct{}{}
-	g.adjList[vet2][vet1] = struct{}{}
+	g.adjList[vet1] = append(g.adjList[vet1], vet2)
+	g.adjList[vet2] = append(g.adjList[vet2], vet1)
 }
 
 /* 删除边 */
-func (g *graphAdjList) removeEdge(vet1 vertex, vet2 vertex) {
+func (g *graphAdjList) removeEdge(vet1 Vertex, vet2 Vertex) {
 	_, ok1 := g.adjList[vet1]
 	_, ok2 := g.adjList[vet2]
 	if !ok1 || !ok2 || vet1 == vet2 {
 		panic("error")
 	}
-	// 删除边 vet1 - vet2, 借助 delete 来删除 map 中的键
-	delete(g.adjList[vet1], vet2)
-	delete(g.adjList[vet2], vet1)
+	// 删除边 vet1 - vet2
+	DeleteSliceElms(g.adjList[vet1], vet2)
+	DeleteSliceElms(g.adjList[vet2], vet1)
 }
 
 /* 添加顶点 */
-func (g *graphAdjList) addVertex(vet vertex) {
+func (g *graphAdjList) addVertex(vet Vertex) {
 	_, ok := g.adjList[vet]
 	if ok {
 		return
 	}
 	// 在邻接表中添加一个新链表
-	g.adjList[vet] = make(map[vertex]struct{})
+	g.adjList[vet] = make([]Vertex, 0)
 }
 
 /* 删除顶点 */
-func (g *graphAdjList) removeVertex(vet vertex) {
+func (g *graphAdjList) removeVertex(vet Vertex) {
 	_, ok := g.adjList[vet]
 	if !ok {
 		panic("error")
 	}
 	// 在邻接表中删除顶点 vet 对应的链表
 	delete(g.adjList, vet)
 	// 遍历其它顶点的链表，删除所有包含 vet 的边
-	for _, set := range g.adjList {
-		// 操作
-		delete(set, vet)
+	for _, list := range g.adjList {
+		DeleteSliceElms(list, vet)
 	}
 }
 
 /* 打印邻接表 */
 func (g *graphAdjList) print() {
 	var builder strings.Builder
 	fmt.Printf("邻接表 = \n")
-	for k, v := range g.adjList {
-		builder.WriteString("\t\t" + strconv.Itoa(k.val) + ": ")
-		for vet := range v {
-			builder.WriteString(strconv.Itoa(vet.val) + " ")
+	// 给 g.adjList 排序一下
+	keys := make([]Vertex, 0, len(g.adjList))
```

Comment:
We should not sort the adjList here because it originally preserves the inserting order(Like a queue).

Here is the Java print code:

```java
    /* 打印邻接表 */
    public void print() {
        System.out.println("邻接表 =");
        for (Map.Entry<Vertex, List<Vertex>> entry : adjList.entrySet()) {
            List<Integer> tmp = new ArrayList<>();
            for (Vertex vertex : entry.getValue())
                tmp.add(vertex.val);
            System.out.println(entry.getKey().val + ": " + tmp + ",");
        }
    }
```

Reply:
In go, map entry does not maintain the sequence in which it is inserted.  Code has been modified to be consistent with java